administrator/components/com_fields/src/Model/FieldModel.php

```diff
@@ -1108,6 +1109,37 @@ function () use ($component, $section) {
             }
         }
 
+        $componentInterface = Factory::getApplication()->bootComponent($component);
+
+        if ($componentInterface instanceof FieldsServiceInterface) {
+            $componentInterface->prepareForm($form, $data);
+        } else {
+            // Try to find the component helper.
+            $eName = str_replace('com_', '', $component);
+            $path  = Path::clean(JPATH_ADMINISTRATOR . "/components/$component/helpers/fields.php");
+
+            if (file_exists($path)) {
+                $cName = ucfirst($eName) . ucfirst($section) . 'HelperFields';
+
+                \JLoader::register($cName, $path);
+
+                if (class_exists($cName) && \is_callable([$cName, 'onPrepareForm'])) {
+                    $lang->load($component, JPATH_BASE, null, false, false)
+                        || $lang->load($component, JPATH_BASE . '/components/' . $component, null, false, false)
+                        || $lang->load($component, JPATH_BASE, $lang->getDefault(), false, false)
+                        || $lang->load($component, JPATH_BASE . '/components/' . $component, $lang->getDefault(), false, false);
+                    \call_user_func_array([$cName, 'onPrepareForm'], [&$form]);
+
+                    // Check for an error.
+                    if ($form instanceof \Exception) {
+                        $this->setError($form->getMessage());
+
+                        return false;
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
        }
```

no reason to support non-bootable components anymore for new features.

Reply:
True!